### PR TITLE
OCPBUGS-20034: improve on-demand data gathering timing issues

### PR DESCRIPTION
**Pull request by tremes**
This extracts the "listener" for the on-demand data gathering requests/resources to separate function. Why should this be better than the previous solution?

- the periodic data gathering doesn't block (when running) the on-demand data gathering requests
- new context is created only when a new valid dataGather resource is created - this used to have unexpected consequences - e.g the on-demand datagathering request didn't work because the context was already timeout

## Categories

- [X] Bugfix
- [ ] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

- no new data gathered

## Documentation

- 

## Unit Tests

- simple unit test added, but it will be better to test in the integration tests

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

No

## References

https://issues.redhat.com/browse/OCPBUGS-20034
https://access.redhat.com/solutions/???

### CODE REVIEW

**Comment by JoaoFula** [comment]
Verification is on my side RN. Will provide qe-approved label only after verification due to broken qe-approved/OCPBUGS process (see [OCPCRT-268](https://issues.redhat.com/browse/OCPCRT-268))

**Review by ncaak — APPROVED** [review]
/lgtm

**Comment by tremes** [comment]
We have some flaky and unrelated tests. Overriding for now
/override ci/prow/insights-operator-e2e-tests

**Pull request merged**
